### DISCUSSION TITLE
Help with a tricky overflow & horizontal scroll layout

### DISCUSSION THREAD

**Discussion by handodk** [tailwindlabs/tailwindcss#9803] [Help] [answered]
Hello,

I've made this Play to try and describe my dilemma: https://play.tailwindcss.com/t6y6NQUETs

I'm trying to build a horizontal slider made up of individual "cards" where the first card always lines up with the left margin of the centred container div (which has a fixed max width and mx-auto). But then I'd like to allow the div containing the cards to "spill out" (overflow) the container div so it can stretch all the way to the right-hand edge of the browser.

The tricky part is getting the left  margin on the div that wraps the "cards" to stick to the left margin of the overall container div.

Any help would be much appreicated.

Thanks.

**Comment by wongjn** [comment, marked as answer]
Use `max(0%, 50% - (theme(maxWidth.lg) / 2))` to calculate the left-side margin: https://play.tailwindcss.com/PFN7jzvkWM

The `50% - (theme(maxWidth.lg) / 2)` calculates the half the difference in width between the parent width and the desired `max-w-lg` width. We then clamp it to `0%` using the `max()` function, so that the `margin-left` will be `0` if the former calculation would be negative for narrower viewports.

**Reply by handodk** [reply]
Thank you thank you thank you! This is great. Worked a treat.
